internal/service.go

```diff
@@ -352,8 +359,23 @@ func (h *HauserService) WriteBundleToJson(bundleID int, filename string) (bytesW
 	return written, nil
 }
 
+func (h *HauserService) getValueConverter() warehouse.ValueToStringFn {
+	if h.config.StorageOnly {
+		return warehouse.ValueToString
+	}
+	return h.database.ValueToString
+}
+
+func (h *HauserService) lastSyncPoint(ctx context.Context) (time.Time, error) {
```

Comment:
Ehh, I could go either way. I kinda like the semantics of the `Storage` and `Database` embedding since it requires that anything that implements the interface is also `Syncable`, but I'm not stuck on that formulation. If we intend to allow more complicated behavior, like doing a sync locally, but still uploading to a database, then making it first class could be useful. Other than that, I'm not sure what it buys us.